Update the documentation framework

ethomson opened the pull request:
This is a backport of #1915 to the `v6` branch.

Some updates to the documentation framework (the scripts that build and display docs) to both make them a little shinier and easier to work with, and to help align these docs with the npm registry documentation.  (The npm registry documentation includes these docs as a subset, so there are some changes to make them more easily consumable.)

1. Drop Gatsby.  For our needs, it's big and slow, and it produces complex content.  Introduce a new application "dockhand" that takes our markdown content, converts that to HTML, puts that in a simple template, and does some minor translations.  The output is standalone HTML files with zero external dependencies.

2. An update to the `docs-build.js` manpage generation script to parse data out of the frontmatter of the documentation pages, instead of parsing the documentation content.  Moving this to metadata makes the content focused on the display, and the frontmatter is focused on the metadata.

3. Update the documentation itself to move the metadata into frontmatter.

4. Move the commands directory to `commands` (instead of `cli-commands`).  Since they're in the `cli` repo, it's clear that they're cli commands.  😀 

5. Fix a few broken links, and a few typos.

Aside from (5), there are no changes to the documentation content itself.

ethomson committed:
docs: use frontmatter for man page generation

Update the `docs-build.js` script to parse the name, man page section,
and description out of the frontmatter and use that for the generated
man page.  This helps avoid repetition between the frontmatter section
and the body of the page itself.

Update the docs pages themselves to remove the H1 (title) and H2
(description).  Finally, ensure that the frontmatter begins in column
0 on each page.

ethomson committed:
docs: reformat code block

Gatsby's markdown parser doesn't like code blocks spread over multiple
lines.

ethomson committed:
docs: use commands not cli-commands

Avoid needless and unnecessary redundancy.

ethomson committed:
docs: change command links to `/commands`

These links now reflect the reality of the filesystem and their paths.

ethomson committed:
docs: correct additional links

Update links to reflect where the pages actually are.

darcyclarke reviewed (changes requested):
We need to update the reference to `cli-commands` in [`./lib/help.js`](https://github.com/npm/cli/blob/release/v7.0.0/lib/help.js#L148) so we don't break `npm help`.

I sent you some other feedback internally about the other issues with routing when viewing the docs statically but generally, the docs that are bundled with the CLI should be able to be viewed/navigated, I think, without requiring a server to be spun up/down.

ethomson committed:
docs: update path to commands

ethomson committed:
docs: use "dockhand" for static generation

Drop Gatsby, use a simplified custom static site generator that uses GFM
and a template for extremely lightweight docs.

ethomson committed:
docs: publish `output` dir to netlify

ethomson committed:
docs: include a table of contents

isaacs committed:
Merge docs deps with main project

This also reduces the publish file size rather considerably.

ethomson committed:
docs: tidy up generation script

Conform to the norm.

ethomson commented:
@darcyclarke I updated this based on how #1915 landed and included 77ad86b5eedf139dda3329a6686d5f104dc233bb
